pandas/io/stata.py

```diff
@@ -820,7 +846,7 @@ def __eq__(self, other: Any) -> bool:
         )
 
     @classmethod
-    def get_base_missing_value(cls, dtype):
+    def get_base_missing_value(cls, dtype: np.dtype) -> Union[int, float]:
```

Comment:
```suggestion
    def get_base_missing_value(cls, dtype: np.dtype) -> float:
```

Reply:
I think we should keep the Union here as well; it doesn't make a difference to mypy it does make a difference to the reader

Reply:
This is enough of a nudge for me to go with what I also like better